src/coreclr/gc/gc.cpp

```diff
@@ -25767,7 +25767,7 @@ void gc_heap::calculate_new_heap_count ()
 
                     dprintf (6666, ("we want to grow but DATAS is limiting, trigger a gen2 right away"));
 #ifdef BACKGROUND_GC
-                    if (background_running_p())
+                    if (is_bgc_in_progress())
```

Comment:
what is the difference between these checks?

Reply:
this is the impl of `is_bgc_in_progress` - 

```cplusplus
BOOL gc_heap::is_bgc_in_progress()
{
#ifdef MULTIPLE_HEAPS
    // All heaps are changed to/from the bgc_initialized state during the VM suspension at the start of BGC,
    // so checking any heap will work.
    gc_heap* hp = g_heaps[0];
#else
    gc_heap* hp = pGenGCHeap;
#endif //MULTIPLE_HEAPS

    return (background_running_p() || (hp->current_bgc_state == bgc_initialized));
}
```
so it includes `background_running_p()` and `(hp->current_bgc_state == bgc_initialized)` which checks for that eph GC.